sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,164 @@
+		}
+	}
+
+	/**
+	 * Return the public key for this passkey.
+	 */
+	getPublicKey(): PublicKey {
+		return new PasskeyPublicKey(this.publicKey);
+	}
+
+	/**
+	 * Return the signature for the provided data (i.e. blake2b(intent_message)).
+	 * This is sent to passkey as the challenge field.
+	 */
+	async sign(data: Uint8Array) {
+		// sents the passkey to sign over challenge as the data.
+		const credential = (await navigator.credentials.get({
+			publicKey: {
+				challenge: data,
+				userVerification: 'required',
+				timeout: 60000,
+			},
+		})) as AuthenticationCredential;
+
+		// parse authenticatorData (as bytes), clientDataJSON (decoded as string).
+		const authenticatorData = new Uint8Array(credential.response.authenticatorData);
+		const clientDataJSON = new Uint8Array(credential.response.clientDataJSON); // response.clientDataJSON is already UTF-8 encoded JSON
+		const decoder = new TextDecoder();
+		const clientDataJSONString: string = decoder.decode(clientDataJSON);
+
+		// parse the signature from DER format, normalize and convert to compressed format (33 bytes).
+		const sig = secp256r1.Signature.fromDER(new Uint8Array(credential.response.signature));
+		let normalized = sig.normalizeS().toCompactRawBytes();
+		const compressedPubkey = secp256r1.ProjectivePoint.fromHex(toHex(this.publicKey)).toRawBytes(
+			true,
+		);
+
+		if (
+			normalized.length !== PASSKEY_SIGNATURE_SIZE ||
+			compressedPubkey.length !== PASSKEY_PUBLIC_KEY_SIZE
+		) {
+			throw new Error('Invalid signature or public key length');
+		}
+
+		// construct userSignature as flag || sig || pubkey for the secp256r1 signature.
+		const arr = new Uint8Array(1 + normalized.length + compressedPubkey.length);
+		arr.set([SIGNATURE_SCHEME_TO_FLAG['Secp256r1']]);
+		arr.set(normalized, 1);
+		arr.set(compressedPubkey, 1 + normalized.length);
+
+		// serialize all fields into a passkey signature according to https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md#signature-encoding
+		return PasskeyAuthenticator.serialize({
+			authenticatorData: authenticatorData,
+			clientDataJson: clientDataJSONString,
+			userSignature: arr,
+		}).toBytes();
+	}
+
+	/**
+	 * This overrides the base class implementation that accepts the raw bytes and signs its
+	 * digest of the intent message, then serialize it with the passkey flag.
+	 */
+	async signWithIntent(bytes: Uint8Array, intent: IntentScope): Promise<SignatureWithBytes> {
+		// prepend it into an intent message and computes the digest.
+		const intentMessage = messageWithIntent(intent, bytes);
+		const digest = blake2b(intentMessage, { dkLen: 32 });
+
+		// sign the digest.
+		const signature = await this.sign(digest);
+
+		// prepend with the passkey flag.
+		const serializedSignature = new Uint8Array(1 + signature.length);
+		serializedSignature.set([SIGNATURE_SCHEME_TO_FLAG[this.getKeyScheme()]]);
+		serializedSignature.set(signature, 1);
+		return {
+			signature: toBase64(serializedSignature),
```

Comment:
in the next major release, it might be good to make this the default signWithIntent, and move the appending of publicKey into sign rather than signWithIntent for the other keypairs so we can keep things consistent, but this is fine for

Reply:
the default signwithintent uses `toSerializedSignature` that does more more than i need (flag || sig || pk) so i override it. you mean to override it for other schemes so they can share the common part? 

https://github.com/MystenLabs/sui/blob/489f42152bd2bdcc5ba4021a7bfca1e44fc7e09d/sdk/typescript/src/cryptography/keypair.ts#L42C21-L42C42

Reply:
nothing to change in this PR, more something I think can be improved down the line to make things more consistent